File: C GCC 5.4.1/ABC/ABC001/ABC001_B_v01.c
// AtCoder Beginner Contest 001 B
// [URL]      https://atcoder.jp/contests/abc001/tasks/abc001_2
// [compiler] C (GCC 5.4.1)
// 
#include <stdio.h>

int main (void)
{
	/* variables for main process */
	int visibility_inpt;
	int visibility_otpt;

	// STEP.01
	// read the values of visibility [m]
	scanf( "%d", &visibility_inpt );

	// STEP.02
	// calculate the visibility and output it
	if      (     0 <= visibility_inpt && visibility_inpt <    100 ) { printf( "%2.2d\n", 0 ); }
	else if (   100 <= visibility_inpt && visibility_inpt <=  5000 ) { printf( "%2.2d\n", visibility_inpt / 100 ); }
	else if (  6000 <= visibility_inpt && visibility_inpt <= 30000 ) { printf( "%2.2d\n", visibility_inpt / 1000 + 50 ); }
	else if ( 35000 <= visibility_inpt && visibility_inpt <= 70000 ) { printf( "%2.2d\n", ( visibility_inpt / 1000 -30 ) / 5 + 80 ); }
	else                                                             { printf( "%2.2d\n", 89 ); };

	// STEP.TRUE_END
	return 0;
}

File: Haskell/ABC/ABC001/ABC001_B_v01.hs
-- [contest]  ABC 001
-- [task]     B
-- [URL]      https://atcoder.jp/contests/abc001/tasks/abc001_2
-- [compiler] Haskell (GHC 7.10.3)
-- [status]   https://atcoder.jp/contests/abc001/submissions/6408045 : WA
--            https://atcoder.jp/contests/abc001/submissions/6410694 : WA
--            https://atcoder.jp/contests/abc001/submissions/6410698 : WA
--            https://atcoder.jp/contests/abc001/submissions/6410707 : WA
--            https://atcoder.jp/contests/abc001/submissions/6410728 : AC

-- modules to import --

import qualified Data.ByteString.Char8
import qualified Data.Maybe
import qualified Text.Printf
import           Prelude

-- functions for this task --

read_Integer = Prelude.fst . Data.Maybe.fromJust . Data.ByteString.Char8.readInteger
get_Integer  = read_Integer <$> Data.ByteString.Char8.getLine

convert_ToKiloMeter :: Integer -> Integer
convert_ToKiloMeter val_meter = Prelude.div (val_meter) (1000)

calculate_visibility :: Integer -> Integer
calculate_visibility val_meter
        | val_meter <    100 = 0
        | val_meter <=  5000 = convert_ToKiloMeter $ val_meter * 10
        | val_meter <= 30000 = (convert_ToKiloMeter $ val_meter) + 50
        | val_meter <= 70000 = (Prelude.div ((convert_ToKiloMeter $ val_meter) - 30) (5)) + 80
        | Prelude.otherwise  = 89

-- the main process is as follows --

main :: IO ()
main = do

        -- STEP.01
        -- read out the given data
        val_meter <- get_Integer

        -- STEP.02
        -- calculate & output the answer of this task
        Text.Printf.printf "%2.2d\n" (calculate_visibility $ val_meter)

-- EOF --
